src/middlewares/authentication.js

```diff
@@ -33,11 +35,19 @@ export default {
 
     if (blacklistedToken) throw new ApplicationError(403, 'Please login or signup to access this resource');
 
-    jwt.verify(token, SECRET_KEY, (error, decodedToken) => {
-      if (error) throw new ApplicationError(401, `${error.message}`);
+    jwt.verify(token, SECRET_KEY, async (error, decodedToken) => {
+      if (error) return next(new ApplicationError(401, `${error.message}`));
```

Comment:
Why was `throw` changed to return? What benefit does this give us? I know express automatically uses the error middleware when `next()` is called with an argument, what benefit does this give us over simply throwing the error which does the same thing?